### PR TITLE
Updated Stop auto-follow on scroll

### PR DESCRIPTION
**Pull request by aditygrg2**
Added support for stopping events when scrolled, and re-starting on clicking on an button. 

Issue : https://github.com/glaslos/ochi/issues/20

### CODE REVIEW

**Comment by aditygrg2** [comment]
Hello @glaslos, please check this PR, and let me know if I understood the requirement correctly

File: src/App.svelte [hunk 1]
```diff
@@ -17,12 +17,22 @@
...
+	let showEvents: boolean;
+
 	export let messages: messageType[] = [];
 
 	// truncate the number of messages show in the app
 	$: if (messages.length >= 50) {
 		messages = messages.slice(1);
 	}
+	
+	$: eventStatus = showEvents;
```
Comment (glaslos) [inline src/App.svelte:30]: Why are you using this proxy variable?

File: src/App.svelte [hunk 2]
```diff
@@ -17,12 +17,22 @@
...
+
+	$: if(eventStatus){
```
Comment (glaslos) [inline src/App.svelte:32]: We don't want the event stream to stop, just auto follow be disabled for a period.

**Review by glaslos — CHANGES_REQUESTED** [review]

**Comment by aditygrg2** [comment]
Understood. I will update this PR accordingly

**Comment by aditygrg2** [comment]
Hi @glaslos,

I have tried these things while solving this problem -

1. I tried to control the scroll events through javascript, by analyzing that if the user is at the bottom (scrolled max), then I am updating the scroll values by maximum (scrollHeight - clientHeight), else the scroll value does not change and thus position remains intact. But this didn't work, as Svelte is updating changes but it is not considering and saving the scrolls, it is showing the element into view always.

2. In other frameworks like react, we can pass keys so that the re-rendering effect does not change or modify other properties of DOM, but I think in svelte passing keys is only limited to adding/removing elements and rendering them only from DOM and not modifying other properties

3. I tried modifying scroll-anchor property but still no success, as svelte was still scrolling as soon as the new event appears.

Can you please help me with any ideas or suggestions using which we can implement this?

**Comment by glaslos** [comment]
@aditygrg2 I will give it a try after work. Sounds odd that this is a problem with Svelte 🤔

**Comment by glaslos** [comment]
Have a look how it can be solve in #22

**Comment by aditygrg2** [comment]
Yes, got it, it was simple, I was going the tough way.

**Pull request closed without merging**